src/EditorFeatures/Test/Workspaces/ProjectCacheHostServiceFactoryTests.cs

```diff
@@ -246,11 +250,38 @@ protected internal override HostWorkspaceServices CreateWorkspaceServices(Worksp
             }
         }
 
+        private class MockOptionService : IWorkspaceOptionService
+        {
+            private readonly OptionSet _optionSet = new WorkspaceOptionSet(null);
+
+#pragma warning disable CS0067 // unused event
+            public event EventHandler<OptionChangedEventArgs> OptionChanged;
+#pragma warning restore CS0067
+
+            public T GetOption<T>(Option<T> option) => option.DefaultValue;
+
+            public T GetOption<T>(PerLanguageOption<T> option, string languageName) => option.DefaultValue;
+
+            public object GetOption(OptionKey optionKey) => optionKey.Option.DefaultValue;
+
+            public OptionSet GetOptions() => _optionSet;
+
+            public IEnumerable<IOption> GetRegisteredOptions() => Enumerable.Empty<IOption>();
+
+            public Task<OptionSet> GetUpdatedOptionSetForDocumentAsync(Document document, OptionSet optionSet, CancellationToken cancellationToken) => SpecializedTasks.FromResult(_optionSet);
+            public void OnWorkspaceDisposed(Workspace workspace) { }
+
+            public void RegisterDocumentOptionsProvider(IDocumentOptionsProvider documentOptionsProvider) { }
+
+            public void SetOptions(OptionSet optionSet) { }
+        }
+
         private class MockHostWorkspaceServices : HostWorkspaceServices
         {
             private readonly HostServices _hostServices;
             private readonly Workspace _workspace;
             private static readonly IWorkspaceTaskSchedulerFactory s_taskSchedulerFactory = new WorkspaceTaskSchedulerFactory();
+            private static readonly IWorkspaceOptionService s_optionService = new MockOptionService();
```

Comment:
This PR will likely have conflicts with https://github.com/dotnet/roslyn/pull/40289